### DISCUSSION TITLE
RDKit cartridge fingerprints

### DISCUSSION THREAD

**Discussion by aigulkhkmv** [rdkit/rdkit#4241] [Q&A] [answered]
Dear RDKit users and contributors, 

I try to use RDKit cartridge for postgresql database as in tutorial https://www.rdkit.org/docs/Cartridge.html. I've generated fingerprints for a database using an RDKit cartridge.

Now, I want to run similarity search as in tutorial:
```bash
chembl_25=# select count(*) from rdk.fps where mfp2%morganbv_fp('Cc1ccc2nc(-c3ccc(NC(C4N(C(c5cccs5)=O)CCC4)=O)cc3)sc2c1');
```
But I want to run this in `python3` code. 

Is it correct to use ` AllChem.GetMorganFingerprintAsBitVect()` function:
```bash 
curs.execute("select * from rdk.fps where mfp2%%s",
AllChem.GetMorganFingerprintAsBitVect(AllChem.MolFromSmiles('Cc1ccc2nc(-c3ccc(NC(C4N(C(c5cccs5)=O)CCC4)=O)cc3)sc2c1'), radius=2)
```
 to generate fingerprint for search? Maybe I can import `morganbv_fp` function from RDKit? It would be more clear usages, but I can't find where I can import this functions.

Thanks!

**Comment by greglandrum** [comment, marked as answer]
Hi @aigulkhkmv, I did a blog post a while ago which shows how to pass fingerprints from python to the cartridge: http://rdkit.blogspot.com/2017/04/using-custom-fingerprint-in-postgresql.html
Here's some sample code:
```
>>> qry = Chem.MolFromSmiles('Cc1ccc2nc(-c3ccc(NC(C4N(C(c5cccs5)=O)CCC4)=O)cc3)sc2c1')
>>> fp = rdMolDescriptors.GetMorganFingerprintAsBitVect(qry,2,512)
>>> from rdkit import DataStructs
>>> curs.execute('select count(*) from rdk.fps where mfp2%%bfp_from_binary_text(%s)',(DataStructs.BitVectToBinaryText(fp),))
>>> curs.fetchall()
[(70,)]
```
To make sure it's working correctly, here I generate the same FP with the cartridge:
```
>>> curs.execute("select count(*) from rdk.fps where mfp2%morganbv_fp('Cc1ccc2nc(-c3ccc(NC(C4N(C(c5cccs5)=O)CCC4)=O)cc3)sc2c1',2)")
>>> curs.fetchall()
[(70,)]
```
Note that it's important that you get the fingerprint length the same; the cartridge uses 512 bits for morgan fingerprints.

**Reply by aigulkhkmv** [reply]
Thanks a lot, Greg!
